pkg/nodeidentity/aws/identify.go

```diff
@@ -101,6 +102,9 @@ func (i *nodeIdentifier) IdentifyNode(ctx context.Context, node *corev1.Node) (*
 	if instanceState != ec2.InstanceStateNameRunning {
 		return nil, fmt.Errorf("found instance %q, but state is %q", instanceID, instanceState)
 	}
+	if instance.InstanceLifecycle != nil {
+		spot = true
+	}
```

Comment:
Seems more like the previous blocks this way.
```suggestion

	spot := false
	if instance.InstanceLifecycle != nil {
		spot = true
	}
```